### PR TITLE
cpu_time intrinsic

### PR DESCRIPTION
**Pull request by vdonaldson**

### CODE REVIEW

File: flang/lib/Lower/IntrinsicCall.cpp [hunk 1]
```diff
@@ -1832,6 +1837,16 @@ IntrinsicLibrary::genCount(mlir::Type resultType,
       });
 }
 
+// CPU_TIME
+void IntrinsicLibrary::genCpuTime(llvm::ArrayRef<fir::ExtendedValue> args) {
+  assert(args.size() == 1);
+  auto arg = args[0].getUnboxed();
```
Comment (jeanPerier) [inline flang/lib/Lower/IntrinsicCall.cpp:1843]: Even though this should be guaranteed by the front-end and lowering, I would still add a small `assert(arg && "time must be a scalar ")` to rule-out any undefined behaviors.

**Review by jeanPerier — APPROVED** [review]
Looks good

File: flang/test/Lower/intrinsic-procedures.f90 [hunk 2]
```diff
@@ -169,6 +169,17 @@ function btest_test(i, j)
   btest_test = btest(i, j)
 end
 
+! CPU_TIME
+! CHECK-LABEL: cpu_time_test
+subroutine cpu_time_test
+  ! CHECK: %[[t:[0-9]+]] = fir.alloca f32 {bindc_name = "t"
+  real :: t
```
Comment (mleair) [inline flang/test/Lower/intrinsic-procedures.f90:176]: I suggest making 't' a dummy argument so there's no chance that the store to 't' gets optimized away.

File: flang/lib/Lower/IntrinsicCall.cpp [hunk 3]
```diff
@@ -431,9 +432,9 @@ struct IntrinsicLibrary {
   mlir::Value genIbits(mlir::Type, llvm::ArrayRef<mlir::Value>);
   mlir::Value genIbset(mlir::Type, llvm::ArrayRef<mlir::Value>);
   fir::ExtendedValue genIchar(mlir::Type, llvm::ArrayRef<fir::ExtendedValue>);
-  mlir::Value genIEOr(mlir::Type, llvm::ArrayRef<mlir::Value>);
+  mlir::Value genIeor(mlir::Type, llvm::ArrayRef<mlir::Value>);
   fir::ExtendedValue genIndex(mlir::Type, llvm::ArrayRef<fir::ExtendedValue>);
-  mlir::Value genIOr(mlir::Type, llvm::ArrayRef<mlir::Value>);
+  mlir::Value genIor(mlir::Type, llvm::ArrayRef<mlir::Value>);
```
Comment (schweitzpgi) [inline flang/lib/Lower/IntrinsicCall.cpp:437]: These renaming games beg the question again. Why `CpuTime` instead of `Cputime`?

**Review by schweitzpgi — APPROVED** [review]

File: flang/lib/Lower/IntrinsicCall.cpp (hunk 3 above)
Comment (vdonaldson) [reply flang/lib/Lower/IntrinsicCall.cpp:437]: It's a minor choice to fit within the now slightly fewer arbitrary variations.

**Pull request merged**